jparest/src/main/java/uk/gov/homeoffice/digital/sas/jparest/controller/ResourceApiController.java

```diff
@@ -233,17 +233,14 @@ public ApiResponse<T> update(@RequestParam UUID tenantId, @PathVariable UUID id,
     payload.setId(id);
     this.entityValidator.validateAndThrowIfErrorsExist(payload);
 
-    var transactionDefinition = new DefaultTransactionDefinition();
-    var transactionStatus =
-        this.transactionManager.getTransaction(transactionDefinition);
-
     var orig = repository.findById(id).orElseThrow(() -> new ResourceNotFoundException(id));
 
     validateResourceTenantId(tenantId, orig, id);
 
-    BeanUtils.copyProperties(payload, orig, EntityUtils.ID_FIELD_NAME);
-
+    var transactionDefinition = new DefaultTransactionDefinition();
+    var transactionStatus = this.transactionManager.getTransaction(transactionDefinition);
     try {
+      BeanUtils.copyProperties(payload, orig, EntityUtils.ID_FIELD_NAME);
```

Comment:
I'm curious about this line. It has the potential to throw a `BeansException` which I **think** can happen if it's not able to copy properties from source to target (maybe an invalid/unexpected field).
In this case would the transaction still roll back? Could we have a similar test to assert no idle transaction in the event that exception occurs?

Tbh I would've thought it would get rolled back as it's a subclass of RuntimeException which we're catching and rolling back. But then again so is the `ResourceNotFoundException` and that didn't seem to get caught before the change.

Reply:
yeah an exception raised by copyProperties should now get caught by the try/catch RuntimeException and rolled back

good idea to increase the test coverage around it, do you think it's needed for this fix or the upcoming refactor could cover it?

Reply:
Yeah tbh it's sufficient for a workaround, test can easily be left for the upcoming refactor 👍